manager/integration/tests/test_ha.py

```diff
@@ -3060,7 +3059,134 @@ def test_engine_image_not_fully_deployed_perform_dr_restoring_expanding_volume()
     15. In a 2-min retry verify that Longhorn doesn't create new replica
        for vol-1 and doesn't reuse the failed replica on node-x
     """
-    pass
+    tainted_node_id = \
+        prepare_engine_not_fully_deployed_environment(client, core_api)
+
+    # step 1
+    volume1 = create_and_check_volume(client, "vol-1", num_of_replicas=2,
+                                      size=str(1 * Gi))
+
+    # node1: tainted node, node2: self host node, node3: the last one
+    nodes = client.list_node()
+    for node in nodes:
+        if node.id == get_self_host_id():
+            node2 = node
+        elif node.id != tainted_node_id and node.id != get_self_host_id:
+            node3 = node
+
+    # step 2
+    volume1 = volume1.attach(hostId=node2.id)
+    volume1 = wait_for_volume_healthy(client, volume1.name)
+
+    volume_endpoint = get_volume_endpoint(volume1)
+    snap1_offset = 1
+    snap_data_size_in_mb = 4
+    write_volume_dev_random_mb_data(volume_endpoint,
+                                    snap1_offset, snap_data_size_in_mb)
+
+    snap = create_snapshot(client, volume1.name)
+    volume1.snapshotBackup(name=snap.name)
+    wait_for_backup_completion(client,
+                               volume1.name,
+                               snap.name,
+                               retry_count=600)
+    bv, b1 = find_backup(client, volume1.name, snap.name)
+
+    dr_volume_name = volume1.name + "-dr"
+    client.create_volume(name=dr_volume_name, size=str(1 * Gi),
+                         numberOfReplicas=2, fromBackup=b1.url,
+                         frontend="", standby=True)
+    wait_for_volume_creation(client, dr_volume_name)
+    wait_for_backup_restore_completed(client, dr_volume_name, b1.name)
+    dr_volume = client.by_id_volume(dr_volume_name)
+
+    # step 4
+    on_node2 = False
+    on_node3 = False
+    for replica in dr_volume.replicas:
+
+        print()
+        if replica.hostId == node2.id:
```

Comment:
Print this dummy print.

Also plz remove unnecessary empty line. Usually we only use empty lines to group the related code for better understanding.

Reply:
Sorry for the wrong code format, I will pay more attention in the future